### DISCUSSION TITLE
How to keep the noise down?

### DISCUSSION THREAD

**Discussion by fregante** [renovatebot/renovate#16622] [Request Help]
I’ve always hated the noise that dependency-update bots caused in repos and I kept distance from all of them for years. 

I usually update all the dependencies at once manually and do light maintenance in one go, every few months. This week I thought Renovate’s “update all dependencies” PRs could be of help here.

These single PRs are great but Renovate’s behavior is still incredibly frustrating even in this mode. I read that this is due to Renovate’s “stateless” approach, but that doesn’t excuse this behavior:

- immortal PRs and issues

I cannot overstate how terrible of an experience that is. 

- I close a “Dependency dashboard” issue, Renovate reopens it without comment (even if there’s nothing to update)
- I close an “Update dependencies” PR and Renovate creates a brand new, identical one

I assume that this mode of operation is not the focus of Renovate because I don’t think anyone actually using it would be ok with bots actively undoing their actions. 

As a default, it’s awful, and I can’t find a way to “fix it” even with configuration.

**Comment by rarkins** [comment]
Hi @fregante, we're very much open to finding ways to keep the "noise" down for people like yourself.

Regarding Dependency Dashboard, it's enabled as part of the config:base preset but it's easily disabled by config. I think you may find though that the dashboard is a good way to reduce noise itself. For example, you can then choose to gate any major update behind dashboard approval and suppress the majority until you're ready.

Regarding "all updates", what is your expectation when you close it? I can assume from your remarks that you expect it to stay closed, but for how long? I assume you are not closing it because you decided "I never want updates for this repo again". So then would your expectation be that it reopens or is recreated as soon as there's a change in updates available? Is that really a desirable outcome for you either? "Close it until something changes" doesn't seem all that noise reducing. Finally, what's your reason for closing - broken tests, or because you think the updates are not worth it?

**Reply by fregante** [reply]
To clarify: I'd like Renovate to open a PR every 5 months. That's it. That's what my configuration appears to ask. Every other operation seems to be unexpected (even though it "makes sense" in Renovate’s current logic)

> I think you may find though that the dashboard is a good way to reduce noise itself.

I do like the dashboard, but this issue serves no purpose and must be (automatically) closed: https://github.com/fregante/daily-version-action/issues/22

I merged the "update all" PR (https://github.com/fregante/daily-version-action/pull/23) and manually closed the issue. Renovate reopened it. Why?

My understanding is that the issue was opened because on the first run it found dependencies to be updated but the scheduled run is still a few months away. Opening it was unexpected, but welcome.

> Regarding "all updates", what is your expectation when you close it?

As per my clarification, if my schedule demands one PR every 5 months, I would never expect any more PRs to ever be created. In this specific instance, at a minimum, it could compare the PR it's about to open to one that already exists, and avoid opening it. If in 5 months there are still no changes, then still _do not_ open another, identical PR.

Low-noise means ≤3 PRs per year. Is there a confutation that successfully does that and only that? 

The rest isn't applicable, but I'll answer it:

> "Close it until something changes" doesn't seem all that noise reducing

The alternative is "Reopen what the user just closed"? I'd take "Close it until something changes" any day.

> Finally, what's your reason for closing - broken tests, or because you think the updates are not worth it?

In my specific case was that I merged a Renovate PR and then reverted it because it caused issues. So Renovate reopened one. So I closed it because I didn't want to deal with it right away, I'd pick up the old PR anyway. Then Renovate opened a third one. 😣

**Comment by PhilipAbed** [comment]
@fregante  sorry for the inconvenience, i appreciate your kind approach on this, 

if you want renovate to run every 3 months for example, check this out https://docs.renovatebot.com/presets-schedule/#schedulequarterly

if there's some dependency you dont want to show up, add a PackageRule to exclude it, 
or if you are waiting for a major patch for example, add a PackageRule for a major patch only, 
or [restrict versions](https://docs.renovatebot.com/configuration-options/#allowedversions)

Dependency Dashboard can be disabled easily using: https://docs.renovatebot.com/presets-schedule/#schedulequarterly

Note that Closing PRs or Dependency Dashboard will not help,
instead make sure Renovate run doesn't find those dependencies you don't want updated anymore,
Renovate will destroy the PR/Dashboard and rebuild it as long as it finds any dependency to work on.

**Reply by PhilipAbed** [reply]
another way to keep the noise is instead of having `all dependencies`.
use restriction on the number of open prs [prConcurrentLimit](https://docs.renovatebot.com/configuration-options/#prconcurrentlimit),
now if there's a PR you dont like, you can just close it, and it will be considered Ignored, 
it won't reopen since its not an immortal PR, that way you can see ALL dependencies in the Dashboard, but you wont have more than 1 or 2 or a few open PRs depending on your `prConcurrentLimit` config, unless you checkbox in the dashboard and  force it open ofcourse.

Renovate's approach is to reduce noise as much as it can
